client/src/index.jsx

```diff
@@ -14,13 +15,34 @@ const Wrapper = Styled.div`
 class App extends React.Component {
   constructor() {
     super();
+    
+    this.state = {
+      locations: [],
+      materials: []
+    }
+    this.getProductById = this.getProductById.bind(this);
   }
   
+  getProductById() {
```

Comment:
same review as the last one here
   - use promise.all
   - call setState once
   - see which window.location variable is conducive now that we're using react hashRouter